builds/misc/images.yaml

```diff
@@ -9,263 +9,409 @@ pr: none
 variables:
   NugetSecurityAnalysisWarningLevel: warn
 
-jobs:
+stages:
 ################################################################################
-  - job: linux_dotnet_projects
+- stage: BuildExecutables
 ################################################################################
-    timeoutInMinutes: 180
-    displayName: LinuxDotnet
-    pool:
-      name: $(pool.linux.name)
-      demands:
-        - ImageOverride -equals agent-aziotedge-ubuntu-18.04-docker
+  displayName: Build Excecutables
+  pool:
+    name: $(pool.linux.name)
+    demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-18.04-docker  
+  dependsOn: []
+  jobs:
+  - job: BuildDotnetComponents
+    displayName: Build Dotnet Components
     steps:
-      - task: Docker@2
-        displayName: Docker login edgebuilds
-        inputs:
-          command: login
-          containerRegistry: iotedge-edgebuilds-acr
-
-      - script: scripts/linux/buildBranch.sh -c $(Build.Configuration) --no-rocksdb-bin
-        name: build
-        displayName: Build ($(Build.Configuration)) dotnet artifacts
-
-      - template: templates/build-broker-watchdog.yaml
-
-      - script: scripts/linux/createArtifactInfo.sh --output-folder '$(Build.BinariesDirectory)/publish' --build-number $(Build.BuildNumber)
-        displayName: 'Create Artifact Info File'
-
-      - task: PublishBuildArtifacts@1
-        displayName: 'Publish Artifacts'
-        inputs:
-          PathtoPublish: '$(Build.BinariesDirectory)/publish'
-          ArtifactName: 'core-linux'
-
-      # Edge Agent
-      - template: templates/image-linux.yaml
-        parameters:
-          name: Edge Agent
-          imageName: azureiotedge-agent
-          project: Microsoft.Azure.Devices.Edge.Agent.Service
+    - script: scripts/linux/buildBranch.sh -c $(Build.Configuration) --no-rocksdb-bin
+      name: build
+      displayName: Build ($(Build.Configuration)) dotnet artifacts
+    - task: PublishBuildArtifacts@1
+      displayName: Publish Dotnet Artifacts
+      inputs:
+        PathtoPublish: '$(Build.BinariesDirectory)/publish'
+        artifactName: 'dotnet_artifacts'
+        StoreAsTar: true
```

Comment:
Why is this 'StoreAsTar' option needed now to preserve Unix file permissions? It wasn't necessary prior to these changes.

This is a documented known issue with the [Publish/Download]BuildArtifacts task but it's interesting that we're only seeing it now: https://github.com/Microsoft/azure-pipelines-tasks/issues/6364

Reply:
I'd prefer to not have to use `storeAsTar` because it makes it impossible to use the `itemPattern` filter in the `DownloadBuildArtifacts` task, thus requiring download of the entire (in this case 500 MB) artifact.

Reply:
It turns out that this bug with dropping execute permissions on artifacts has been occurring all along. The reason it did not cause problems in the past is because nothing was consuming the executables in the core-linux artifact. Instead, building of executables and images was happening in the same job (i.e. on the same agent) so the executables could be copied into the appropriate directories prior to building the images. Now however, the executables are passed as artifacts between stages, so the altered execute permissions are causing problems.